libcaf_net/caf/net/interfaces.hpp

```diff
@@ -0,0 +1,44 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2018 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+#include "caf/net/ip.hpp"
+#include "caf/optional.hpp"
+
+namespace caf {
+namespace net {
+
+/// Utility class bundling access to network interface names and addresses.
+class interfaces {
+public:
+  /// Returns a native IPv4 or IPv6 translation of `host`.
+  static optional<std::pair<std::string, ip>>
+  native_address(const std::string& host, optional<ip> preferred = none);
+
+  /// Returns the host and protocol available for a local server socket
+  static std::vector<std::pair<std::string, ip>>
+  server_address(uint16_t port, const char* host,
```

Comment:
In the past, we needed this function for `publish` to work with a port only. However, we want to ditch that design anyway. Can't we drop this "guess-magic" entirely and instead encode everything we need in the config via URI? That's code I'd really like to get rid of.